### PR TITLE
update configuring ansible-galaxy client

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

Update how to configure ansible-galaxy client before using it.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs.ansible.com
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: docs/docsite/rst/shared_snippets/galaxy_server_list.txt [hunk 1]
```diff
@@ -1,11 +1,19 @@
 
 
+If you use the ``ansible-galaxy`` command, you need to first configure the server lists that command communicates with. You can configure this as follows:
```
Comment (jborean93) [inline docs/docsite/rst/shared_snippets/galaxy_server_list.txt]: You don't need to do this if using the `https://galaxy.ansible.com/`, only really for things like Automation Hub or dev/custom Galaxy instances. That's why I had the link to `:ref:galaxy_server` as that's the default server to use.

File: docs/docsite/rst/shared_snippets/galaxy_server_list.txt [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
-following entries like so:
+* Set the server list in the :ref:`galaxy_server_list` configuration option in :ref:`ansible_configuration_settings_locations` file.
+* Use the ``--server`` command line argument to limit to an individual server.
+
+To configure a Galaxy server list:
+
+#. Add the ``server_list``  option under the ``[galaxy]`` section to one or more server names.
+#. Create a new section for each server name.
+#. Set the ``url`` option for each server name.
+#. Set the ``auth_url`` option for each server name.
```
Comment (jborean93) [inline docs/docsite/rst/shared_snippets/galaxy_server_list.txt:18]: Only needed for Automation Hub (no idea what the URL looks like though)

File: docs/docsite/rst/shared_snippets/galaxy_server_list.txt [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
-argument when it performs an action against a Galaxy server. The ``ansible-galaxy collection install`` supports
-installing collections from multiple servers as defined in the :ref:`ansible_configuration_settings_locations` file
-using the :ref:`galaxy_server_list` configuration option. To define multiple Galaxy servers you have to create the
-following entries like so:
+* Set the server list in the :ref:`galaxy_server_list` configuration option in :ref:`ansible_configuration_settings_locations` file.
+* Use the ``--server`` command line argument to limit to an individual server.
+
+To configure a Galaxy server list:
+
+#. Add the ``server_list``  option under the ``[galaxy]`` section to one or more server names.
```
Comment (jborean93) [inline docs/docsite/rst/shared_snippets/galaxy_server_list.txt:12]: Does `#.` mean a bullet list as well?

File: docs/docsite/rst/shared_snippets/galaxy_server_list.txt [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
+#. Create a new section for each server name.
+#. Set the ``url`` option for each server name.
+#. Set the ``auth_url`` option for each server name.
+#. Set the authentication for each server name. Use either the ``token`` option or the ``username`` and ``password`` options.
```
Comment (jborean93) [inline docs/docsite/rst/shared_snippets/galaxy_server_list.txt]: I think `username` `password` is the older method for Automation Hub. Not sure if that ever made it into AH so maybe check with @alikins to see if we should still document it

**Review by jborean93 — COMMENTED** [review]
Just a few comments everything looks good though

File: docs/docsite/rst/user_guide/collections_using.rst [hunk 5]
```diff
@@ -17,6 +17,16 @@ You can install and use collections through `Ansible Galaxy <https://galaxy.ansi
 Installing collections
 ======================
 
+.. _galaxy_server_config:
+
+Configuring the ``ansible-galaxy`` client
+------------------------------------------
+
+.. include:: ../shared_snippets/galaxy_server_list.txt
```
Comment (newswangerd) [inline docs/docsite/rst/user_guide/collections_using.rst]: Can we update the snippet to use the real automation hub URLs?

They should be
```
url=https://cloud.redhat.com/api/automation-hub/
auth_url=https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token
```

File: docs/docsite/rst/shared_snippets/galaxy_server_list.txt (hunk 3 above)
Comment (samccann) [reply docs/docsite/rst/shared_snippets/galaxy_server_list.txt:12]: It's rst fancy for a numbered list :-)  The output looks like this:
image

Comment (jborean93) [reply docs/docsite/rst/shared_snippets/galaxy_server_list.txt:12]: Ahh cool, TIL

**Comment by chouseknecht** [comment]
Can we make this simpler?

We have basically two questions that we need to answer: 

- What do I do to download from Community Galaxy, and what do I do to publish to Community Galaxy?

- What do I do to download from Automation Hub, and what do I do if I want to publish to Automation Hub? 

IMHO, what's in our docs and in this PR is a  bunch of technical foo that doesn't answer these questions with simple, clear examples.

**Review by chouseknecht — CHANGES_REQUESTED** [review]
See comment above.

**Comment by jborean93** [comment]
No idea how to best translate it into actual docs but the answers to your questions @chouseknecht are

> What do I do to download from Community Galaxy, and what do I do to publish to Community Galaxy?

Nothing, `ansible-galaxy` is set to use `https://galaxy.ansible.com/` by default

> What do I do to download from Automation Hub, and what do I do if I want to publish to Automation Hub?

Configure the server list in `ansible.cfg` with the AH URL and auth values.

File: docs/docsite/rst/user_guide/collections_using.rst (hunk 5 above)
Comment (acozine) [reply docs/docsite/rst/user_guide/collections_using.rst]: Updated in 95d3a456bb08b198564bb3e0232a2aa45f286db2.

**Comment by chouseknecht** [comment]
> No idea how to best translate it into actual docs but the answers to your questions @chouseknecht are
> 
> > What do I do to download from Community Galaxy, and what do I do to publish to Community Galaxy?
> 
> Nothing, `ansible-galaxy` is set to use `https://galaxy.ansible.com/` by default
> 
That might be true for downloading content from Community Galaxy. Great! Let's say that somewhere, clearly. It's however not true to publish to Community Galaxy. In that case you need an API token. How do you get one? Where in the config do you put it? Let's explains that and show an example.

> > What do I do to download from Automation Hub, and what do I do if I want to publish to Automation Hub?
> 
> Configure the server list in `ansible.cfg` with the AH URL and auth values.

Great! Let's show a clear and simple example of that with the correct URLs.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 6]
```diff
@@ -271,12 +271,26 @@ You can publish collections to Galaxy using the ``ansible-galaxy collection publ
 
 .. note:: Once you upload a version of a collection, you cannot delete or modify that version. Ensure that everything looks okay before you upload it.
 
+.. _galaxy_get_token:
+
+Getting your token or API key
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To upload your collection to Galaxy, you must first obtain an API token (``--api-key`` in the ``ansible-galaxy`` CLI command). The API token is a secret token used by Ansible Galaxy to protect your content.
```
Comment (chouseknecht) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: To upload your collection to Galaxy or Automation Hub, you must first obtain an API token (``--api-key`` in the ``ansible-galaxy`` CLI command). The API token is a secret token used ~~by Ansible Galaxy~~ to protect your content.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 7]
```diff
@@ -271,12 +271,26 @@ You can publish collections to Galaxy using the ``ansible-galaxy collection publ
...
+
+To get your API token:
+
+* For galaxy, go to `Galaxy profile preferences <https://galaxy.ansible.com/me/preferences>`_ page and click :guilabel:`API token`.
```
Comment (chouseknecht) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: * For Galaxy, visit the `Galaxy profile preferences <https://galaxy.ansible.com/me/preferences>`_ page and click :guilabel:`API token`.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 8]
```diff
@@ -271,12 +271,26 @@ You can publish collections to Galaxy using the ``ansible-galaxy collection publ
...
+* for Automation Hub, go to https://cloud.redhat.com/ansible/automation-hub/token/ and click :guilabel:`Get API token` from the version dropdown.
```
Comment (chouseknecht) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: * For Automation Hub, go to https://cloud.redhat.com/ansible/automation-hub/token/ and click :guilabel:`Get API token` from the version dropdown.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 9]
```diff
@@ -271,12 +271,26 @@ You can publish collections to Galaxy using the ``ansible-galaxy collection publ
...
+
 .. _upload_collection_ansible_galaxy:
 
 Upload using ansible-galaxy
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-See :ref:`Configuring the ansible-galaxy client <collections_installing>` to set up your Galaxy server list before trying to publish a collection to Galaxy.
+
+.. note::
+  By default, ``ansible-galaxy`` uses https://galaxy.ansible.com as the Galaxy server (as listed in the :file:`ansible.cfg` file under :ref:`galaxy_server`). You do not need any further configuration. See :ref:`Configuring the ansible-galaxy client <galaxy_server_config>` if you are using any other Galaxy server, such as Red Hat Automation Hub).
```
Comment (chouseknecht) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: By default, ``ansible-galaxy`` uses https://galaxy.ansible.com as the Galaxy server (as listed in the :file:`ansible.cfg` file under :ref:`galaxy_server`). If you are publishing your collection to Galaxy, you do not need any further configuration. See :ref:`Configuring the ansible-galaxy client <galaxy_server_config>` if you are using any other server, such as Red Hat Automation Hub.

**Comment by chouseknecht** [comment]
Made a few minor comments with suggested edits. Overall, this is looking a lot better.

**Review by chouseknecht — APPROVED** [review]

**Pull request merged**